### PR TITLE
Reset reason support

### PR DESCRIPTION
**Pull request by Lurkerpas**

### CODE REVIEW

File: src/CbiPartitionApi.c [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+Reset_Reason g_reset_reason = Reset_Reason_Unknown;
+
+/**
+ * @brief Get the Reset Canary address
+ *
+ * @return Reset canary address
+ */
+static unsigned char* getResetCanaryAddress()
+{
+    return (char*)((uintptr_t)(&__end)
```
Comment (Ficzer) [inline src/CbiPartitionApi.c:58]: Shouldn't it be unsigned char?

File: src/CbiPartitionApi.c [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+/**
+ * @brief Returns whether Reset Canary is set
+ *
+ * @return Whether Reset Canary is set
+ */
+static bool isResetCanarySet()
+{
+    unsigned volatile char* buffer = getResetCanaryAddress();
```
Comment (Ficzer) [inline src/CbiPartitionApi.c:69]: Shouldn't it be unsigned char?

File: src/CbiPartitionApi.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    return true;
+}
+
+/**
+ * @brief Set the Reset Canary
+ *
+ */
+static void setResetCanary()
+{
+    unsigned volatile char* buffer = getResetCanaryAddress();
```
Comment (Ficzer) [inline src/CbiPartitionApi.c:84]: Shouldn't it be unsigned char?

File: src/CbiPartitionApi.c (hunk 1 above)
Comment (Lurkerpas) [reply src/CbiPartitionApi.c:58]: Yes

File: src/CbiPartitionApi.c (hunk 2 above)
Comment (Lurkerpas) [reply src/CbiPartitionApi.c:69]: Yes

File: src/CbiPartitionApi.c (hunk 3 above)
Comment (Lurkerpas) [reply src/CbiPartitionApi.c:84]: Yes

**Pull request merged**